### ISSUE TITLE
eqFTP not listing Directory

### ISSUE THREAD

**Issue by ChiefWahoo80** [Equals182/eqFTP#194]
I make a new connection and eqFTP say the it is connected to the Server but I do not see any directories. I use eqFTP Version 0.7.2. 
Thats the error:

```
[node-log 3:53:19 PM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /home/chiefwahoo80/
NodeDebugUtils.js:117 [node-log 3:53:19 PM] [eqFTP-ftpDomain]: [q.p] Can't get folder: /home/chiefwahoo80/. Error: {"code":"ETIMEDOUT","errno":"ETIMEDOUT","syscall":"connect"}
NodeDebugUtils.js:117 [node-log 3:53:19 PM] [eqFTP-ftpDomain]: Disconnecting...
main.js:1629 Object {event: "server_disconnecting", connectionID: 2}
2NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: [c.c] ftpDomain currently disconnecting from: 2
NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: Disonnected.
main.js:1629 Object {event: "server_disconnect", connectionID: 2}
main.js:1629 Object {event: "working", work: false}
NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: [q.p] Queue is empty
NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /home/chiefwahoo80/
NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: [q.p] Got folder: /home/chiefwahoo80/
main.js:1629 Object {event: "directory_got", err: false, path: "", connectionID: 2, queuer: Object}
ErrorNotification.js:125 Exception in 'eqFTP:events' listener on NodeConnection {domains: Object, _registeredModules: Array[1], _pendingInterfaceRefreshDeferreds: Array[0], _pendingCommandDeferreds: Array[6], _autoReconnect: true…} TypeError: Cannot read property 'length' of undefined TypeError: Cannot read property 'length' of undefined
    at Function.n.extend.each (file:///C:/Program%20Files%20(x86)/Brackets/www/thirdparty/thirdparty.min.js:556:2727)
    at file:///C:/Users/Philipp/AppData/Roaming/Brackets/extensions/user/equals182.eqftp/main.js:1666:23
    at NodeConnection.trigger (/utils/EventDispatcher.js:222:40)
    at Object.triggerWithArray (/utils/EventDispatcher.js:260:28)
    at NodeConnection._receive (/utils/NodeConnection.js:479:29)
/utils/EventDispatcher.js:225 Assertion failed:
main.js:1629 Object {event: "working", work: false}
NodeDebugUtils.js:117 [node-log 3:53:20 PM] [eqFTP-ftpDomain]: [q.p] Queue is empty
```

Maybe you know what the problem could be? Thanks for help, Phil

**Comment by Equals182** [comment]
Looks like for some reason eqftp gets undefined object instead of list of files. Can you please update from `https://github.com/Equals182/Buggy-eqFTP` and post console log again? There should be a line that starts with "what is this? does it have length property?"

**Comment by ChiefWahoo80** [comment]
Hi, thanks for your answer. i installed from buggy-ftp and here is What is this? and the hole log:

what is this? does it have length property? undefined

Log:
 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: [q.p] Can't get folder: /. Error: {"code":"ETIMEDOUT","errno":"ETIMEDOUT","syscall":"connect"}
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: Disconnecting...
main.js:1629 Object {event: "server_disconnecting", connectionID: 2}
2NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: [c.c] ftpDomain currently disconnecting from: 2
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: Disonnected.
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: [q.p] Queue is empty
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: [q.p] Got folder: /
NodeDebugUtils.js:117 [node-log 9:17:52 AM] [eqFTP-ftpDomain]: [q.p] Queue is empty
main.js:1629 Object {event: "server_disconnect", connectionID: 2}
main.js:1629 Object {event: "working", work: false}
main.js:1629 Object {event: "directory_got", err: false, path: "", connectionID: 2, queuer: Object}
main.js:1666 what is this? does it have length property? undefined
ErrorNotification.js:125 Exception in 'eqFTP:events' listener on NodeConnection {domains: Object, _registeredModules: Array[1], _pendingInterfaceRefreshDeferreds: Array[0], _pendingCommandDeferreds: Array[6], _autoReconnect: true…} TypeError: Cannot read property 'length' of undefined TypeError: Cannot read property 'length' of undefined
    at Function.n.extend.each (file:///C:/Program%20Files%20(x86)/Brackets/www/thirdparty/thirdparty.min.js:556:2727)
    at file:///C:/Users/Philipp/AppData/Roaming/Brackets/extensions/user/equals182.eqftp/main.js:1667:23
    at NodeConnection.trigger (/utils/EventDispatcher.js:222:40)
    at Object.triggerWithArray (/utils/EventDispatcher.js:260:28)
    at NodeConnection._receive (/utils/NodeConnection.js:479:29)
/utils/EventDispatcher.js:225 Assertion failed:
main.js:1629 Object {event: "working", work: false}

**Comment by Equals182** [comment]
Sorry for late response. Please try using different `folder retrieving method` in Connection's settings (Advanced tab)
If this won't help, please update from `https://github.com/Equals182/Buggy-eqFTP` again and look for `what about this` string.

**Comment by vinnygambiny** [comment]
I have the same issue... Here my log with the Buggy-eqFTP

 [eqftp-DEBUG] Click event
main.js:1605 [eqftp-DEBUG] Not disabled. Reading settings
main.js:1607 [eqftp-DEBUG] Settings read.
main.js:1609 [eqftp-DEBUG] Settings: true
main.js:1610 [eqftp-DEBUG] Showith fileTree
main.js:1584 [eqftp-save-debug] ftp.queue.add started
main.js:1591 [eqftp-save-debug] Adding to node queue:  Object {type: "folder", connectionID: 1, path: "", id: "f_1461797615267478", state: "opened"…}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: 1874: [q.a] Queuer is NOT ignored, continuing
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: 1905: [q.a] Queue updated: {"a":[{"type":"folder","connectionID":1,"path":"","id":"f_1461797615267478","state":"opened","queue":"a"}],"p":[],"f":[],"s":[]}
main.js:1645 Object {event: "queue_update", connectionID: 1, element: Object}
main.js:1645 Object {event: "working", work: true}
main.js:1645 Object {event: "queue_update", connectionID: 1, element: Object}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: Connecting to this ID: 1
main.js:1645 Object {event: "server_connecting", connectionID: 1}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: Connecting...
main.js:1645 Object {event: "server_connect", connectionID: 1}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: Connected...
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:35 PM] {
  "code": 220,
  "text": "220---------- Welcome to Pure-FTPd [privsep] [TLS] ----------\n220-You are user number 2 of 50 allowed.\n220-Local time is now 18:53. Server port: 21.\n220-This is a private system - No anonymous login\n220-IPv6 connections are also welcome on this server.\n220 You will be disconnected after 15 minutes of inactivity.",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:35 PM] "user ****"
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:35 PM] {
  "code": 331,
  "text": "331 User **** OK. Password required",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:35 PM] "pass ****"
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:35 PM] {
  "code": 230,
  "text": "230 OK. Current restricted directory is /",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:35 PM] "feat"
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:35 PM] {
  "code": 211,
  "text": "211-Extensions supported:\n EPRT\n IDLE\n MDTM\n SIZE\n MFMT\n REST STREAM\n MLST type*;size*;sizd*;modify*;UNIX.mode*;UNIX.uid*;UNIX.gid*;unique*;\n MLSD\n AUTH TLS\n PBSZ\n PROT\n UTF8\n TVFS\n ESTA\n PASV\n EPSV\n SPSV\n ESTP\n211 End.",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: [s.gRR] Getting remote root for this ID: 1
NodeDebugUtils.js:117 [node-log 6:53:35 PM] [eqFTP-ftpDomain]: [c.c] Connection already exists for this ID: 1
NodeDebugUtils.js:117 [node-log 6:53:35 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:35 PM] "pwd"
NodeDebugUtils.js:117 [node-log 6:53:36 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:36 PM] {
  "code": 257,
  "text": "257 \"/\" is your current location",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:36 PM] [eqFTP-ftpDomain]: [s.gRR] Remote root for this ID: 1 is /
NodeDebugUtils.js:117 [node-log 6:53:36 PM] [eqFTP-ftpDomain]: [q.p] Connected: /
NodeDebugUtils.js:117 [node-log 6:53:36 PM] [eqFTP-ftpDomain]: [q.p] Queuer type: folder
NodeDebugUtils.js:117 [node-log 6:53:36 PM] [eqFTP-ftpDomain]: [q.p] Getting path: 
NodeDebugUtils.js:117 [node-log 6:53:36 PM] [eqFTP-ftpDomain]: [c.c] Connection already exists for this ID: 1
NodeDebugUtils.js:117 [node-log 6:53:36 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:36 PM] "TYPE I"
NodeDebugUtils.js:117 [node-log 6:53:36 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:36 PM] {
  "code": 200,
  "text": "200 TYPE is now 8-bit binary",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:53:36 PM] DEBUG:  user_command
NodeDebugUtils.js:117 [node-log 6:53:36 PM] "PASV"
NodeDebugUtils.js:117 [node-log 6:53:36 PM] DEBUG:  server_response
NodeDebugUtils.js:117 [node-log 6:53:36 PM] {
  "code": 227,
  "text": "227 Entering Passive Mode (172,31,15,203,49,50)",
  "isMark": false,
  "isError": false
}
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /
NodeDebugUtils.js:117 [node-log 6:54:51 PM] what about this? { [Error: connect ETIMEDOUT] code: 'ETIMEDOUT', errno: 'ETIMEDOUT', syscall: 'connect' } undefined
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: [q.p] Can't get folder: /. Error: {"code":"ETIMEDOUT","errno":"ETIMEDOUT","syscall":"connect"}
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: Disconnecting...
main.js:1645 Object {event: "server_disconnecting", connectionID: 1}
2NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: [c.c] ftpDomain currently disconnecting from: 1
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: Disonnected.
main.js:1645 Object {event: "server_disconnect", connectionID: 1}
main.js:1645 Object {event: "working", work: false}
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: [q.p] Queue is empty
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: 1180: [s.gP] Got Directory: /
NodeDebugUtils.js:117 [node-log 6:54:51 PM] what about this? false undefined
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: [q.p] Got folder: /
main.js:1645 Object {event: "directory_got", err: false, path: "", connectionID: 1, queuer: Object}
/utils/EventDispatcher.js:235 Exception in 'eqFTP:events' listener on NodeConnection {domains: Object, _registeredModules: Array[1], _pendingInterfaceRefreshDeferreds: Array[0], _pendingCommandDeferreds: Array[24], _autoReconnect: true…} TypeError: Cannot read property 'length' of undefined TypeError: Cannot read property 'length' of undefined
    at Function.n.extend.each (file:///Applications/Brackets.app/Contents/www/thirdparty/thirdparty.min.js:556:2727)
    at file:///Users/vincent/Library/Application%20Support/Brackets/extensions/user/equals182.eqftp/main.js:1682:23
    at NodeConnection.trigger (/utils/EventDispatcher.js:233:40)
    at Object.triggerWithArray (/utils/EventDispatcher.js:271:28)
    at NodeConnection._receive (/utils/NodeConnection.js:479:29)
/utils/EventDispatcher.js:236 Assertion failed:
main.js:1645 Object {event: "working", work: false}
NodeDebugUtils.js:117 [node-log 6:54:51 PM] [eqFTP-ftpDomain]: [q.p] Queue is empty

**Comment by Equals182** [comment]
### Hello there!

I know this issue been there for a pretty long time, but I have some good news here!
We just released eqFTP v**1.0.0-beta.0** (v1-b for short I guess) and we're completely discontinuing support of all previous versions of this plugin.
New version works great on Brackets 1.8 and it should be good on lower versions too.

So yeah, we're closing and locking this issue forever and looking forward for you to try out our new version of eqFTP.
I'm sorry if I somehow disturbed you, or you lost your interest in eqFTP or Brackets in general, we're not going to make any noise here again.

If you have any questions, please contact me via email (you can find it in [my profile](https://github.com/Equals182)) or [Telegram app](http://t.me/equals182)

Thank you for understanding 😊
